### PR TITLE
storcon_cli: add 'drain' command

### PR DESCRIPTION
**Pull request by VladLazar**
## Problem
We need the ability to prepare a subset of storage controller managed pageservers for decommisioning. The storage controller cannot currently express this in terms of scheduling constraints (it's a pretty special case, so I'm not sure it even should).

## Summary of Changes
A new `drain` command is added to `storcon_cli`. It takes a set of nodes to drain and migrates primary attachments outside of said set. Simple round robing assignment is used under the assumption that nodes outside of the draining set are evenly balanced.

Note that secondary locations are not migrated. This is fine for staging, but the migration API will have to be extended for prod in order to allow migration of secondaries as well.

I've tested this out against a neon local cluster. The immediate use for this command will be to migrate staging to ARM(Arch64) pageservers.

Related https://github.com/neondatabase/cloud/issues/14029

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

File: control_plane/storcon_cli/src/main.rs [hunk 1]
```diff
@@ -148,6 +149,22 @@ enum Command {
...
+        nodes: Vec<NodeId>,
+        // Optional: migration concurrency (default is 8)
+        #[arg(long)]
+        concurrency: Option<usize>,
+        // Optional: maximum number of shards to migrate
+        #[arg(long)]
+        max_shards: Option<usize>,
+        // Optional: when set to true, nothing is migrated, but the plan is printed to stdout
+        #[arg(long)]
+        dryrun: Option<bool>,
```
Comment (arpad-m) [inline control_plane/storcon_cli/src/main.rs:166]:
```suggestion
        dry_run: Option<bool>,
```

File: control_plane/storcon_cli/src/main.rs [hunk 2]
```diff
@@ -737,6 +754,197 @@ async fn main() -> anyhow::Result<()> {
...
+                        .expect("We only migrate attached tenant shards"),
+                    to: node_to_fill_descs[selected_node_idx].id,
+                });
+                selected_node_idx = (selected_node_idx + 1) % node_to_fill_descs.len();
+            }
+
+            let total_moves = moves.len();
+
+            if dryrun == Some(true) {
+                println!("Dryrun requested. Planned {} moves:", total_moves);
```
Comment (arpad-m) [inline control_plane/storcon_cli/src/main.rs:884]:
```suggestion
                println!("Dryrun requested. Planned {total_moves} moves:");
```

**Review by arpad-m — APPROVED** [review]

File: control_plane/storcon_cli/src/main.rs (hunk 1 above)
Comment (VladLazar) [reply control_plane/storcon_cli/src/main.rs]: https://github.com/neondatabase/neon/pull/8007/commits/5d902ffb81f1d5757459ea9d0ec9474200df5cc5

File: control_plane/storcon_cli/src/main.rs (hunk 2 above)
Comment (VladLazar) [reply control_plane/storcon_cli/src/main.rs]: https://github.com/neondatabase/neon/pull/8007/commits/9ca62a50fd6708604f5feee442521284cb6ad277

**Pull request merged**